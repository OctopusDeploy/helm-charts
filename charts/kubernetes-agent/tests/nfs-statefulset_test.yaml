suite: "persistence"
templates:
- templates/nfs-statefulset.yaml
tests:
- it: "is not created when storageClassName has a value"
  set:
    persistence:
      storageClassName: "value"
  asserts:
  - hasDocuments:
      count: 0

- it: "should match snapshot"
  asserts:
  - matchSnapshot: {}

- it: sets the emptyDir size to be the value in values
  set:
    persistence:
      size: 100Gi
  asserts:
  - equal:
      path: spec.template.spec.volumes[?(@.name == 'octopus-volume')].emptyDir.sizeLimit
      value: 100Gi