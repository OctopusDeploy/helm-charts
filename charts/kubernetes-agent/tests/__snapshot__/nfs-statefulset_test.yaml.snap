should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: StatefulSet
    metadata:
      labels:
        app.kubernetes.io/name: octopus-agent-nfs
      name: octopus-agent-nfs
      namespace: NAMESPACE
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/name: octopus-agent-nfs
      template:
        metadata:
          labels:
            app.kubernetes.io/name: octopus-agent-nfs
        spec:
          containers:
            - env:
                - name: SHARED_DIRECTORY
                  value: /octopus
                - name: SYNC
                  value: "true"
              image: octopusdeploy/nfs-server:1.0.1
              imagePullPolicy: IfNotPresent
              name: octopus-agent-nfs
              ports:
                - containerPort: 2049
              resources:
                requests:
                  cpu: 50m
                  memory: 50Mi
              securityContext:
                privileged: true
              volumeMounts:
                - mountPath: /octopus
                  name: octopus-volume
          nodeSelector:
            kubernetes.io/os: linux
          terminationGracePeriodSeconds: 30
          volumes:
            - emptyDir:
                sizeLimit: 1Gi
              name: octopus-volume
